WordPress/src/main/java/org/wordpress/android/ui/uploads/PostUploadNotifier.java

```diff
@@ -309,31 +318,57 @@ void updateNotificationError(PostModel post, SiteModel site, String errorMessage
-//        if (notificationData.notificationErrorId == 0) {
-//            notificationData.notificationErrorId = notificationData.notificationId + (new Random()).nextInt();
-//        }
         doNotify(errorNotificationId, notificationBuilder.build());
     }
 
-    void updateNotificationProgress(PostModel post, float progress) {
+    void updateNotificationProgressForMedia(MediaModel media, float progress) {
+        if (sNotificationData.totalMediaItems == 0 && sNotificationData.totalPostItems == 0) {
+            return;
+        }
 
-        //TODO MEDIA reimplement although most probably will REMOVE THIS COMPLETELY
-//        NotificationData notificationData = sPostIdToNotificationData.get(post.getId());
-//        if (notificationData.totalMediaItems == 0) {
-//            return;
-//        }
-//
-//        // Simple way to show progress of entire post upload
-//        // Would be better if we could get total bytes for all media items.
-//        double currentChunkProgress = (notificationData.itemProgressSize * progress);
-//
-//        if (notificationData.currentMediaItem > 1) {
-//            currentChunkProgress += notificationData.itemProgressSize * (notificationData.currentMediaItem - 1);
-//        }
-//
-//        mNotificationBuilder.setProgress(100, (int) Math.ceil(currentChunkProgress), false);
-//        doNotify(sPostIdToNotificationData.get(post.getId()).notificationId, mNotificationBuilder.build());
+        // only update if media item is in our map - this check is performed because
+        // it could happen that a media item is already done uploading but we receive an upload
+        // progress event from FluxC after that. We just need to avoid re-adding the item to the map.
+        Float currentProgress = sNotificationData.mediaItemToProgressMap.get(media.getId());
+        // also, only set updates in increments of 5% per media item to avoid lots of notification updates
+        if (currentProgress != null && progress > (currentProgress + 0.05f)) {
+            setProgressForMediaItem(media.getId(), progress);
+            updateNotificationProgress();
+        }
+    }
+
+    private void updateNotificationProgress() {
+        if (sNotificationData.totalMediaItems == 0 && sNotificationData.totalPostItems == 0) {
+            return;
+        }
+
+        mNotificationBuilder.setProgress(100, (int) Math.ceil(getCurrentOverallProgress() * 100), false);
+        doNotify(sNotificationData.notificationId, mNotificationBuilder.build());
+    }
+
+    private void setProgressForMediaItem(int mediaId, float progress) {
+        sNotificationData.mediaItemToProgressMap.put(mediaId, progress);
+    }
+
+    private float getCurrentOverallProgress() {
+        int totalItemCount = sNotificationData.totalPostItems + sNotificationData.totalMediaItems;
+        float currentMediaProgress = getCurrentMediaProgress();
+        float overAllProgress = 0;
```

Comment:
We don't need to set this to zero first since it's being assigned in the next line.

Reply:
nice! addressed in 17e9d55